src/ECER.Resources.Accounts/PortalInvitations/PortalInvitationRepository.cs

```diff
@@ -0,0 +1,27 @@
+﻿using AutoMapper;
+using ECER.Utilities.DataverseSdk.Model;
+using Microsoft.Xrm.Sdk.Client;
+
+namespace ECER.Resources.Accounts.PortalInvitations;
+
+internal class PortalInvitationRepository : IPortalInvitationRepository
+{
+  private readonly EcerContext context;
+  private readonly IMapper mapper;
+
+  public PortalInvitationRepository(EcerContext context, IMapper mapper)
+  {
+    this.context = context;
+    this.mapper = mapper;
+  }
+
+  public async Task<IEnumerable<PortalInvitation>> Query(PortalInvitationQuery query)
+  {
+    await Task.CompletedTask;
+    var portalInvitations = context.ecer_PortalInvitationSet.Where(pi => pi.ecer_PortalInvitationId == query.portalInvitationId);
```

Comment:
if you're querying a single row, you can join to get the related rows instead of lazy load them

Reply:
I tried that first, but it did not work, I don't know why. just changed it to this and worked

Reply:
it will probably work if you use .Single()